sql/core/src/test/resources/sql-tests/inputs/aggregates_part1.sql

```diff
@@ -0,0 +1,151 @@
+
+-- SQL2003 binary aggregates [SPARK-23907]
+-- SELECT regr_count(b, a) FROM aggtest;
+-- SELECT regr_sxx(b, a) FROM aggtest;
+-- SELECT regr_syy(b, a) FROM aggtest;
+-- SELECT regr_sxy(b, a) FROM aggtest;
+-- SELECT regr_avgx(b, a), regr_avgy(b, a) FROM aggtest;
+-- SELECT regr_r2(b, a) FROM aggtest;
+-- SELECT regr_slope(b, a), regr_intercept(b, a) FROM aggtest;
+SELECT covar_pop(b, a), covar_samp(b, a) FROM aggtest;
+SELECT corr(b, a) FROM aggtest;
+
+
+-- test accum and combine functions directly [SPARK-23907]
+-- CREATE TABLE regr_test (x float8, y float8);
+-- INSERT INTO regr_test VALUES (10,150),(20,250),(30,350),(80,540),(100,200);
+-- SELECT count(*), sum(x), regr_sxx(y,x), sum(y),regr_syy(y,x), regr_sxy(y,x)
+-- FROM regr_test WHERE x IN (10,20,30,80);
+-- SELECT count(*), sum(x), regr_sxx(y,x), sum(y),regr_syy(y,x), regr_sxy(y,x)
+-- FROM regr_test;
+-- SELECT float8_accum('{4,140,2900}'::float8[], 100);
+-- SELECT float8_regr_accum('{4,140,2900,1290,83075,15050}'::float8[], 200, 100);
+-- SELECT count(*), sum(x), regr_sxx(y,x), sum(y),regr_syy(y,x), regr_sxy(y,x)
+-- FROM regr_test WHERE x IN (10,20,30);
+-- SELECT count(*), sum(x), regr_sxx(y,x), sum(y),regr_syy(y,x), regr_sxy(y,x)
+-- FROM regr_test WHERE x IN (80,100);
+-- SELECT float8_combine('{3,60,200}'::float8[], '{0,0,0}'::float8[]);
+-- SELECT float8_combine('{0,0,0}'::float8[], '{2,180,200}'::float8[]);
+-- SELECT float8_combine('{3,60,200}'::float8[], '{2,180,200}'::float8[]);
+-- SELECT float8_regr_combine('{3,60,200,750,20000,2000}'::float8[],
+--                            '{0,0,0,0,0,0}'::float8[]);
+-- SELECT float8_regr_combine('{0,0,0,0,0,0}'::float8[],
+--                            '{2,180,200,740,57800,-3400}'::float8[]);
+-- SELECT float8_regr_combine('{3,60,200,750,20000,2000}'::float8[],
+--                            '{2,180,200,740,57800,-3400}'::float8[]);
+-- DROP TABLE regr_test;
+
+
+-- test count, distinct
+SELECT count(four) AS cnt_1000 FROM onek;
+SELECT count(DISTINCT four) AS cnt_4 FROM onek;
+
+select ten, count(*), sum(four) from onek
+group by ten order by ten;
+
+select ten, count(four), sum(DISTINCT four) from onek
+group by ten order by ten;
+
+-- user-defined aggregates
+-- SELECT newavg(four) AS avg_1 FROM onek;
+-- SELECT newsum(four) AS sum_1500 FROM onek;
+-- SELECT newcnt(four) AS cnt_1000 FROM onek;
+-- SELECT newcnt(*) AS cnt_1000 FROM onek;
+-- SELECT oldcnt(*) AS cnt_1000 FROM onek;
+-- SELECT sum2(q1,q2) FROM int8_tbl;
+
+-- test for outer-level aggregates
+
+-- this should work
+select ten, sum(distinct four) from onek a
+group by ten
+having exists (select 1 from onek b where sum(distinct a.four) = b.four);
+
+-- this should fail because subquery has an agg of its own in WHERE
+select ten, sum(distinct four) from onek a
+group by ten
+having exists (select 1 from onek b
+               where sum(distinct a.four + b.four) = b.four);
+
+-- [SPARK-27769] Test handling of sublinks within outer-level aggregates.
```

Comment:
https://issues.apache.org/jira/browse/SPARK-27769

Reply:
I removed the `correctness` label because we support `Infinity` correctly.
```scala
scala> sql("SELECT CAST('Infinity' AS DOUBLE), CAST('infinity' AS DOUBLE)").show
+------------------------+------------------------+
|CAST(Infinity AS DOUBLE)|CAST(infinity AS DOUBLE)|
+------------------------+------------------------+
|                Infinity|                    null|
+------------------------+------------------------+
```